### PR TITLE
My code review 2 suggestions

### PR DESCRIPTION
**Pull request by spauldvl**
This is a Mobile application that finds users random activities to do based upon different attributes such as, number of participants, price range, and type of activity the user would like to partake in. You can then save the activities you would like to do to your account and check them off as you go.
Code was available on time
The program in very well documented especially in MainActivity
Program compiles
I did a lot of best practices changes, code was very solid and it was difficult to find things to add honestly.
[commit1](https://github.com/tron561/IT3048C-Repo/commit/ac15eea3c569f3ecf1ba1af4665f92f1622661d4)
[commit2](https://github.com/tron561/IT3048C-Repo/commit/7efa1b2408df9883f5d2ae818f125e0df37f4e74)
I learned how to implement the sign in with firebase in MainActivity: private fun signIn(), private fun signInResult(), etc
How to implement a button that leads to a desired result GenerateActivityButton() 
How to use CircularProgressIndicator() if there is a wait time to show user the application is loading

### CODE REVIEW

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 1]
```diff
@@ -40,10 +40,10 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
  * This class represents the main activity for the IdeaStorm app and sets up the UI layout and theme
  */
 class MainActivity : ComponentActivity() {
-    var firebaseUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
-    var selectedParticipantOption: String? = null
-    var selectedTypeOption: String? = null
-    var selectedPriceOption: String? = null
+    private var firebaseUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
```
Comment (spauldvl) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:43]: updated modifiers to private per the suggestions here : https://kotlinlang.org/docs/visibility-modifiers.html#constructors

File: app/src/main/java/com/ideastorm/v25001/MainViewModel.kt [hunk 2]
```diff
@@ -13,15 +13,12 @@ import com.ideastorm.v25001.service.IActivityService
 import kotlinx.coroutines.launch
 
 class MainViewModel(var activityService : IActivityService = ActivityService()) : ViewModel() {
-    var activity : MutableLiveData<Activity> = MutableLiveData<Activity>()
```
Comment (spauldvl) [inline app/src/main/java/com/ideastorm/v25001/MainViewModel.kt:16]: removed extra whitespace if it separates a declaration and its type per suggestions: https://kotlinlang.org/docs/coding-conventions.html#colon

File: app/src/main/java/com/ideastorm/v25001/MainViewModel.kt [hunk 3]
```diff
@@ -13,15 +13,12 @@ import com.ideastorm.v25001.service.IActivityService
...
-    var user : User? = null
+    var activity: MutableLiveData<Activity> = MutableLiveData<Activity>()
+    var user: User? = null
 
-    private lateinit var firestore : FirebaseFirestore
+    private var firestore : FirebaseFirestore = FirebaseFirestore.getInstance()
```
Comment (spauldvl) [inline app/src/main/java/com/ideastorm/v25001/MainViewModel.kt:19]: combined declaration and assignment

File: app/src/main/java/com/ideastorm/v25001/RetrofitClientInstance.kt [hunk 4]
```diff
@@ -6,9 +6,9 @@ import retrofit2.converter.gson.GsonConverterFactory
 object RetrofitClientInstance {
 
     private var retrofit: Retrofit? = null
-    private val BASE_URL = "https://www.boredapi.com/api/"
+    private const val BASE_URL = "https://www.boredapi.com/api/"
```
Comment (spauldvl) [inline app/src/main/java/com/ideastorm/v25001/RetrofitClientInstance.kt:9]: changed value to constant

File: app/src/main/java/com/ideastorm/v25001/dto/User.kt [hunk 5]
```diff
@@ -1,4 +1,4 @@
 package com.ideastorm.v25001.dto
 
-data class User(val uid : String = "", var displayName: String?) {
+data class User(val uid: String = "", var displayName: String? = "") {
```
Comment (spauldvl) [inline app/src/main/java/com/ideastorm/v25001/dto/User.kt:3]: initialized attribute

File: app/src/test/java/com/ideastorm/v25001/ActivityTests.kt [hunk 6]
```diff
@@ -45,7 +45,7 @@ class ActivityTests {
     private fun thenTheActivityShouldHaveThreeParticipants() {
         assertNotNull(activity)
         var hasThreeParticipants = false
-        if (activity.participants == (3)) {
```
Comment (spauldvl) [inline app/src/test/java/com/ideastorm/v25001/ActivityTests.kt:48]: removed unnecessary parenthesis

File: app/src/main/java/com/ideastorm/v25001/MainViewModel.kt [hunk 7]
```diff
@@ -13,15 +13,12 @@ import com.ideastorm.v25001.service.IActivityService
...
 
-    init {
-        firestore = FirebaseFirestore.getInstance()
-        firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
-    }
+    init { firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build() }
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainViewModel.kt:21]: I'd keep this as it were.  
- multiple lines are more readable.
- It's easier to add additional logic, if needed, when the logic currently present is on multiple lines.
- If you get an error with a line number, it's much easier to narrow down the error if each command is on a different line, versus all on one.

File: app/src/main/java/com/ideastorm/v25001/dto/User.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/com/ideastorm/v25001/dto/User.kt:3]: If initialized here, does it still require the null (?) attribute?

On the other hand, is this an item that should be required, when constructing a new User object?

**Comment by discospiff** [comment]
Good discussion.

I recommend the group review these changes, and decide whether or not to merge.

**Pull request closed without merging**